### PR TITLE
[Jackson-3] Address #493, disable `DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES` by default.

### PR DESCRIPTION
**Pull request by JooHyukKim**
addresses #493 

### Note

Once merged, **[JStep-2](https://github.com/FasterXML/jackson-future-ideas/wiki/JSTEP-2)** Wiki should be updated.

### CODE REVIEW

File: src/main/java/tools/jackson/databind/DeserializationFeature.java [hunk 1]
```diff
@@ -110,12 +110,12 @@ public enum DeserializationFeature implements ConfigFeature
      * This setting only takes effect after all other handling
      * methods for unknown properties have been tried, and
      * property remains unhandled.
+     * Enabling this feature means that a {@link DatabindException}
+     * will be thrown if an unknown property is encountered.
      *<p>
-     * Feature is enabled by default (meaning that a
-     * {@link DatabindException} will be thrown if an unknown property
-     * is encountered).
+     * Feature is disabled by default.
```
Comment (cowtowncoder) [inline src/main/java/tools/jackson/databind/DeserializationFeature.java:116]: In this case, we probably do want to mention that the default state was different in Jackson 2.x.

**Comment by cowtowncoder** [comment]
@JooHyukKim Looks good, only had 1 comment that I added. I can change that & update release notes and then merge this.

**Comment by cowtowncoder** [comment]
Merged. Thank you @JooHyukKim !

**Comment by cowtowncoder** [comment]
@JooHyukKim Could you update JSTEP-2 page with change? I updated README, release-notes wiki.

**Comment by JooHyukKim** [comment]
> @JooHyukKim Could you update JSTEP-2 page with change? I updated README, release-notes wiki.

Yes, for sure 👍🏼

**Comment by cowtowncoder** [comment]
@JooHyukKim I think this broke a few tests in downstream packages (dataformats (binary, text, xml), jackson-modules-base etc). The reason it was not obvious was because snapshot builds started failing for 3.x few weeks ago -- I think api token I used got disabled for some reason. I created new one, assigned as secret, and snapshot deploys work again... and now breakage is obvious.

**Comment by cowtowncoder** [comment]
Fixed `jackson-modules-base` (2 tests for Afterburner, Blackbird, copied from jackson-databind).

Failing: 

* `jackson-dataformats-binary`: https://github.com/FasterXML/jackson-dataformats-binary/actions/runs/10087812932 
* `jackson-dataformats-text`: https://github.com/FasterXML/jackson-dataformats-text/actions/runs/10087815044
* `jackson-dataformat-xml`: https://github.com/FasterXML/jackson-dataformat-xml/actions/runs/10087817936
* Scala module: https://github.com/FasterXML/jackson-module-scala/actions/runs/10087826937

(and possibly others but these at least)

**Comment by JooHyukKim** [comment]
Oh okay, thanks! Will try to check on others as well.

**Comment by JooHyukKim** [comment]
I wonder there is a way to alert cascading-build failure back to databind 🤔
Or if it should/shouldn't

**Comment by cowtowncoder** [comment]
@JooHyukKim in theory, there probably would be a way (cascading rebuild case could take some action). But for now email notifications probably need to suffice, as long as they get triggered. Although the problem here was not cascading part but snapshot deploy: both since new snapshot version was not made available and since this prevent cascading builds themselves (but without snapshot deploy they wouldn't have found issues either).

**Comment by cowtowncoder** [comment]
Ok so XML and text-formats fixed. Aside from binary-formats, Scala, 2 more failures:

* JAX-RS providers: https://github.com/FasterXML/jackson-jaxrs-providers/actions/runs/10097258838
* Jakarta-RS providers: https://github.com/FasterXML/jackson-jakarta-rs-providers/actions/runs/10097263912

**Pull request merged**